openedx/core/djangoapps/lang_pref/tests/test_middleware.py

```diff
@@ -231,14 +231,7 @@ def test_preference_update_noop(self):
         # No preference yet, should write to the database
 
         self.assertEqual(get_user_preference(self.user, LANGUAGE_KEY), None)
-
-        # The 'email_marketing' app is installed in the LMS env but not the CMS env. It listens for the
-        # USER_FIELD_CHANGED signal (utils.model_utils) and does a query to check the EmailMarketingConfiguration
-        # table to see if Sailthru integreation is enabled.
-        expected_queries = 6 if 'email_marketing' in settings.INSTALLED_APPS else 5
-        with self.assertNumQueries(expected_queries):
-            self.middleware.process_request(self.request)
-
```

Comment:
This query checking is in openedx.core.djangoapps code - but is referencing configuration at the LMS/CMS level. The query counting was broken and seeing 7 LMS queries under jenkins and 6 LMS queries locally. So I've just removed the query counting altogether, which was violating the dependency layering anyway.

Thoughts?

Reply:
I'm generally not a fan of testing query counts, although it looks like @cpennington originally [added this test](https://github.com/edx/edx-platform/commit/1811f7235f720fbe6090dfd811ce211950f79abb#diff-aa378cdc200a5a6a4adebaa9e1306db1) specifically to make sure that the query counts stayed nice and small.  Maybe check with him to see if it's still useful now that it's getting harder to make the count deterministic?

Reply:
@cpennington How vital is this query count?

Reply:
Typically, I care most about query-counts when trying to prevent/cure an n+1 blowup, or when the queries are being done inside an inner loop somewhere.

I think this count was here because the lang-pref middleware (and all middleware) is called on every request, so it's really important that it stay tight.

What was the query that showed on jenkins but not locally? We have a number of different tools for accounting for things like caching in the tests.

Reply:
It was an extra `EmailMarketingConfiguration.current()` call at the end of the query list. I tracked it down to some configuration difference that I would have fixed via mock/patch - but didn't want to patch something in the `lms` layer from the `openedx` layer.

Reply:
Huh. I wonder if the test could just override the list of `INSTALLED_APPS`, so that it knows it's only testing the `lang-pref` middleware, rather than the whole middleware stack.